spotxBidAdapter2.js

```diff
@@ -0,0 +1,363 @@
+      method: 'POST',
+      url: URL + channelId,
+      data: requestPayload,
+      bidRequest: bidderRequest
+    };
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse, bidderRequest) {
+    const bidResponses = [];
+    const serverResponseBody = serverResponse.body;
+
+    const requestMap = {};
+    if (bidderRequest && bidderRequest.data && bidderRequest.data.imp) {
+      utils._each(bidderRequest.data.imp, imp => requestMap[imp.id] = imp);
+    }
+
+    if (serverResponseBody && utils.isArray(serverResponseBody.seatbid)) {
+      utils._each(serverResponseBody.seatbid, function(bids) {
+        utils._each(bids.bid, function(spotxBid) {
+          const request = requestMap[spotxBid.impid];
+
+          const bid = {
+            requestId: request.video.ext.bidId,
+            currency: serverResponseBody.cur || 'USD',
+            cpm: spotxBid.price,
+            creativeId: spotxBid.crid || '',
+            ttl: 360,
+            netRevenue: true,
+            channel_id: serverResponseBody.id,
+            cache_key: spotxBid.ext.cache_key,
+            video_slot: request.video.ext.videoSlot
+          };
+
+          if (request.video) {
+            bid.vastUrl = '//search.spotxchange.com/ad/vast.html?key=' + spotxBid.ext.cache_key;
+            bid.mediaType = VIDEO;
+            bid.width = spotxBid.w;
+            bid.height = spotxBid.h;
+          }
+
+          if (request.video.ext.ad_unit == 'outstream') {
+            let renderer = Renderer.install({
+              id: 0,
+              url: '//',
+              config: {
+                adText: 'SpotX Outstream Video Ad via Prebid.js',
+                player_width: request.video.ext.player_width,
+                player_height: request.video.ext.player_height,
+                content_page_url: request.video.ext.content_page_url,
+                outstream_static: request.video.ext.outstream_static,
+                ad_mute: request.video.ext.ad_mute
+              }
+            });
+
+            try {
+              renderer.setRender(outstreamRender);
+              renderer.setEventHandlers({
+                impression: function impression() {
+                  return utils.logMessage('SpotX outstream video impression event');
+                },
+                loaded: function loaded() {
+                  return utils.logMessage('SpotX outstream video loaded event');
+                },
+                ended: function ended() {
+                  utils.logMessage('SpotX outstream renderer video event');
+                }
+              });
+            } catch (err) {
+              utils.logWarn('Prebid Error calling setRender or setEve,tHandlers on renderer', err);
+            }
+            bid.renderer = renderer;
```

Comment:
This will only be used if the renderer isn't assigned by the pub... I think that's ok, but we should test that

Reply:
Sure. How do you want to proceed with testing?